From b72408d85f1e514414e08c630812736a01cc4f29 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenko@src.gnome.org>
Date: Sat, 3 Feb 2018 15:36:27 +0100
Subject: [PATCH] tools: use real @PYTHON@ instead of /usr/bin/env

Signed-off-by: Igor Gnatenko <ignatenko@src.gnome.org>
---
 tools/g-ir-tool-template.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/g-ir-tool-template.in b/tools/g-ir-tool-template.in
index edc68a4c..133088a6 100644
--- a/tools/g-ir-tool-template.in
+++ b/tools/g-ir-tool-template.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 # -*- Mode: Python -*-
 # GObject-Introspection - a framework for introspecting GObject libraries
 # Copyright (C) 2008  Johan Dahlin
-- 
2.16.1

